pkg/datarepair/checker/checker.go

```diff
@@ -54,30 +53,35 @@ func NewChecker(pointerdb *pointerdb.Service, sdb statdb.DB, repairQueue queue.R
 		repairQueue: repairQueue,
 		overlay:     overlay,
 		irrdb:       irrdb,
-		limit:       limit,
 		logger:      logger,
+		Loop:        sync2.NewCycle(interval),
 	}
-
-	checker.Interval.SetInterval(interval)
-
 	return checker
 }
 
 // Run the checker loop
 func (checker *Checker) Run(ctx context.Context) (err error) {
 	defer mon.Task()(&ctx)(&err)
 
-	return checker.Interval.Run(ctx, func(ctx context.Context) error {
+	return checker.Loop.Run(ctx, func(ctx context.Context) error {
 		err := checker.IdentifyInjuredSegments(ctx)
 		if err != nil {
 			checker.logger.Error("error with injured segments identification: ", zap.Error(err))
 		}
-		return nil
+		return err
 	})
 }
 
-// Close closes resources
-func (checker *Checker) Close() error { return nil }
+// Pause suspends the Checker loop
+func (checker *Checker) Pause() {
```

Comment:
I think this won't be necessary anymore, since the loop itself has been exposed.